### PR TITLE
Copy trust store to /tmp to work with readonly fs

### LINKED ISSUE #281: Failed to update CACERTS when readOnlyRootFilesystem is set
## Expected Behavior
At startup it should create the new cacert in different folder (like tmp) that can be mounted with write rights

## Current Behavior
Currently it try to modify it in place
WARNING: Unable to add container CA certificates to JVM because /layers/paketo-buildpacks_bellsoft-liberica/jre/lib/security/cacerts is read-only

## Steps to Reproduce

1. Run a builded container with readOnlyRootFilesystem set to true

## Motivations

It's mandatory to use readOnlyRootFilesystem in our env

### PR DESCRIPTION
**Pull request by c0d1ngm0nk3y**
fixes #281 

## Summary

In case the truststore is readonly (readonly filesystem), it is copied to `/tmp` and used from there. That allows loading `cacerts` even with a readonly filesystem.

If `/tmp` is also readonly, it will still not fail, but warn the user about this. This is the previous behaviour.

## Use Cases

Container is started with a readonly filesystem for security reasons.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: helper/openssl_certificate_loader.go [hunk 1]
```diff
@@ -21,26 +21,65 @@ import (
...
+
+	trustStoreFile, err := os.Open(trustStore)
+	if err != nil {
+		return nil, fmt.Errorf("unable to open trust store %s\n%w", trustStore, err)
+	}
+	defer trustStoreFile.Close()
+
+	err = sherpa.CopyFile(trustStoreFile, tempTrustStore)
+	if err != nil {
+		return nil, fmt.Errorf("failed to copy dir (%s, %s)\n%w", trustStore, tempTrustStore, err)
```
Comment (dmikusa) [inline helper/openssl_certificate_loader.go:48]:
```suggestion
		return nil, fmt.Errorf("unable to copy dir (%s, %s)\n%w", trustStore, tempTrustStore, err)
```

This is minor, someone before me started the pattern of error messages starting with "unable ...", we try to keep that going.

File: helper/openssl_certificate_loader.go [hunk 2]
```diff
@@ -21,26 +21,65 @@ import (
...
+	}
+
+	opts := sherpa.AppendToEnvVar("JAVA_TOOL_OPTIONS", " ", fmt.Sprintf("-Djavax.net.ssl.trustStore=%s", tempTrustStore))
```
Comment (dmikusa) [inline helper/openssl_certificate_loader.go:51]: What do you think about checking to see if this setting already exists?

My only hesitation is that I'm not sure we could do anything, aside from log a warning message and continue on.

Thoughts?

**Review by dmikusa — CHANGES_REQUESTED** [review]
Looks good. One very minor thing and one question.

File: helper/openssl_certificate_loader.go (hunk 2 above)
Comment (c0d1ngm0nk3y) [reply helper/openssl_certificate_loader.go:52]: > Thoughts?

We discussed this and would probably rather not check this. For 2 reasons:

1.) What should we do in that case? Import into that one? Probably we should fail in that case.
2.) What would be the reason to have a custom truststore? If you want to bring your custom `cacerts` at runtime, you could do so with a binding.

I think it is too much of an edge case to add some extra logic for this.

**Review by dmikusa — APPROVED** [review]

**Comment by loewenstein** [comment]
@dmikusa is there a release planned for libjvm?

**Comment by tojagrut** [comment]
Hi @dmikusa can you release paketo-buildpacks/adoptium with this new version of libjvm?  I am using adoptium buildpack and it is not using the latest released version of this libjvm library.

Cheers, Jagrut

**Comment by dmikusa** [comment]
@tojagrut I just cut releases for all of our JVM provider buildpacks. If you're using one directly you can get that now. Otherwise, we'll bump the composite to include this fix on Friday as part of the normal release cycle.

@loewenstein SAP Machine buildpack is released & includes this fix, and also the other @c0d1ngm0nk3y patched around dependency comparison.

**Comment by beytularedzheb** [comment]
Hello,
It looks like this change caused some issues in applications where users used to provide -Djavax.net.ssl.trustStore in JAVA_TOOL_OPTIONS. 
As you can see below user provided parameter is ignored because of the appended -Djavax.net.ssl.trustStore=/tmp/truststore at the end of JAVA_TOOL_OPTIONS.
Is there any workaround to fix this?
Screenshot 2023-06-11 at 12 59 45

**Comment by dmikusa** [comment]
@beytularedzheb Please open a new issue, and we can discuss it. Include some details about your use case and why you are setting that flag. Thanks

**Pull request merged**